SlicerLite/SlicerLiteLib/SlicerLiteModuleWidget.py

```diff
@@ -0,0 +1,170 @@
+import qt, slicer
+
+from .ButtonItemDelegate import DeleteButtonItemDelegate, DicomMetadataButtonItemDelegate
+from .DataLoader import DataLoader
+from .EventFilters import DragAndDropEventFilter
+from .UIUtils import UIUtils
+from .SlicerUtils import *
+from .ItemModel import ItemModel, Item
+
+
+class SlicerLiteModuleWidget(qt.QWidget):
+    def __init__(self, parent=None):
+        super(SlicerLiteModuleWidget, self).__init__(parent)
+        qt.QVBoxLayout(self)
+        self._lastOpenedDirectory = ""
+
+        self.data_loader = DataLoader()
+        self.item_table_model = ItemModel()
+        self.item_table_view = qt.QTableView()
+        self.item_table_model.dataChanged.connect(lambda _: self.item_table_view.viewport().update())
+        self.delete_button_item_delegate = DeleteButtonItemDelegate()
+        self.dicom_tags_button_item_delegate = DicomMetadataButtonItemDelegate()
+
+        self.buttons_delegate = [self.delete_button_item_delegate, self.dicom_tags_button_item_delegate]
+
+        # Setup event filter
+        self.filter = DragAndDropEventFilter(slicer.util.mainWindow(), self.load_dicom_directory)
+        slicer.util.mainWindow().installEventFilter(self.filter)
+
+        self.setup_ui()
+
+    def setup_ui(self) -> None:
+        """
+        Create and initialize UI components
+        """
+        self.setup_table_view_layout()
+        self.setup_rendering_layout()
+        self.setup_segmentation_layout()
+        self.layout().addStretch()
+
+    def setup_table_view_layout(self) -> None:
+        """
+        Setup the table view behavior/display
+        """
+        self.item_table_model.setHorizontalHeaderLabels(["Name", "Dicom tags", "Delete"])
+        self.item_table_view.setItemDelegateForColumn(1, self.dicom_tags_button_item_delegate)
+        self.item_table_view.setItemDelegateForColumn(2, self.delete_button_item_delegate)
+        # Hide headers
+        self.item_table_view.horizontalHeader().hide()
+        self.item_table_view.verticalHeader().hide()
+        # Hide table borders
+        self.item_table_view.setFrameStyle(qt.QFrame.NoFrame)
+        self.item_table_view.horizontalHeader().setSectionResizeMode(qt.QHeaderView.Stretch)
+        self.item_table_view.horizontalHeader().resizeSection(1, 32)
+        self.item_table_view.horizontalHeader().resizeSection(2, 32)
+        # Make lines no editabled
+        self.item_table_view.setEditTriggers(qt.QAbstractItemView.NoEditTriggers)
+        # Allow only one selection
+        self.item_table_view.setSelectionMode(qt.QAbstractItemView.SingleSelection)
+
+        self.item_table_view.clicked.connect(self.on_table_view_item_clicked)
+        self.item_table_view.setModel(self.item_table_model)
+
+        self.layout().addWidget(UIUtils.createButton("Load", callback=self.on_click_load_dicom_directory))
+        self.layout().addWidget(qt.QLabel("Loaded volumes:"))
```

Comment:
Label is not visible in current UI once a few volumes have been loaded.

It would probably be clearer to have the Horizontal label + frame to make things clearer to a user.